src/query/storages/fuse/src/io/read/block/block_reader_parquet.rs

```diff
@@ -116,6 +126,70 @@ impl BlockReader {
         deserialized_res
     }
 
+    pub(crate) fn schema_default_vals(
+        data_schema: DataSchema,
+        table: Arc<dyn Table>,
+        ctx: Arc<dyn TableContext>,
+    ) -> Result<Vec<Scalar>> {
+        GlobalIORuntime::instance().block_on(async move {
```

Comment:
why need `async move`?

Reply:
need to call `parse_exprs` function, which call tokio `block_in_place` function to resolve expr:

```
pub fn parse_exprs(
    ctx: Arc<dyn TableContext>,
    table_meta: Arc<dyn Table>,
    unwrap_tuple: bool,
    sql: &str,
) -> Result<Vec<Expr>> {
  // ...
      let exprs = ast_exprs
        .iter()
        .map(|ast| {
            let (scalar, _) =
                *block_in_place(|| Handle::current().block_on(type_checker.resolve(ast, None)))?;
            let expr = scalar.as_expr_with_col_index()?;
            Ok(expr)
        })
        .collect::<Result<_>>()?;
}
```

Reply:
Dangerous block_in_place inside a hot loop.